hayeon/programmers-1/문자열-내-마음대로-정렬하기.js

```diff
@@ -0,0 +1,58 @@
+function solution(strings, n) {
+  strings.sort(function (a, b) {
+    let first = a[n];
+    let second = b[n];
+    if (first === second) {
+      return (a > b) - (b > a);
+    } else {
+      return (first > second) - (second > first);
+    }
+  });
+
+  return strings;
+}
```

Comment:
```suggestion
function solution(strings, n) {
  strings.sort((a, b) => {
    const first = a[n];
    const second = b[n];
    
    if (first === second) {
      return (a > b) - (b > a);
    }
    return (first > second) - (second > first);
  });
  return strings;
}
```
- 고생하셨습니다 👍 여러번 강조하지만 first와 second는 재할당을 하지 않기때문에 `const`로 선언해주세요 ! 
- 그리고 sort와 같은 자바스크립트 고차함수의 인자로 들어가는 콜백함수는 함수 선언문보다 `화살표 함수`가 좋습니다 이는 `this 바인딩 문제`와 관련되어 있어요.
- 여기서 a[n]과 b[n]은 따로 first와 second로 변수 할당 안하셔도 좋을 것 같아요 🙆‍♂️ 오히려 코드가 더 길어지고, `a와 b가 붕뜨는 느낌입니다`

<br />

- 개인적으로 아래와 같은 코드가 깔끔해보입니다.
```js
function solution(strings, n) {
  strings.sort((a, b) => {
    if (a[n] === b[n]) {
      return (a > b) - (b > a);
    }
    return (a[n] > b[n]) - (b[n] > a[n]);
  });
  return strings;
}
```
- 제 코드나, 윤우님 코드 꼭 참고해보세요~

Reply:
헉 넵!!! 감사합니다 꼭 다시 풀어보겠습니당!!🧐